### PR TITLE
Chin sgw 64 invalid password bug

### PR DESCRIPTION
**Pull request by chinokoronkwo**
Wallet:
-added console log to catch error raised during switch

Login:
-redirect user to auth root after logout to enable login/logout testing
-fixed image ref
-added login status comments
-removed error  ref no longer being used
-added node-wifi for disconnect tests
-updated divs to provide long status comments
-moved setInvalid login hook to outside if login error block

Tests:
-test successful login w/correct key
-test unsuccessful login w/incorrect key
-test successful login w/correct key and no wifi
-test unsuccessful login w/incorrect key and no wifi

Reviewer
@AlexBrandes

### CODE REVIEW

**Comment by chinokoronkwo** [comment]
Reviewer @AlexBrandes

File: source/scenes/common/Login/Login.tsx [hunk 1]
```diff
@@ -114,17 +120,16 @@ const Login = ({
...
-            {errors.password ? (
-              <span className={errorClass}>{errors.password.message}</span>
+          <div id={'ui-login-status'} className={styles.errorWrapper}>
+            {isInvalid ? (
+              <span id={'ui-login-error'} className={errorClass}>{failureComment}</span>
             ) : (
-              isInvalid && (
-                <span className={errorClass}>Error: Invalid password</span>
-              )
+              <span id={'ui-login-success'} className={errorClass}>{successComment}</span>
```
Comment (AlexBrandes) [inline source/scenes/common/Login/Login.tsx:127]: This shows all the time - even before any action is taken. Let's remove this message completely.
Screen Shot 2022-01-11 at 10 59 49 AM

File: source/scenes/common/Login/Login.tsx [hunk 2]
```diff
@@ -114,17 +120,16 @@ const Login = ({
...
             )}
           </div>
         </div>
         <ButtonV3
+          id={'ui-login-submit-button'}
```
Comment (AlexBrandes) [inline source/scenes/common/Login/Login.tsx:132]: Please update id format to match existing style in other parts of the codebase.

**Review by AlexBrandes — CHANGES_REQUESTED** [review]

File: package.json [hunk 3]
```diff
@@ -55,6 +55,7 @@
     "react-dynamic-font": "^2.0.1"
   },
   "devDependencies": {
+    "node-wifi": "^2.0.16"
```
[the remark this answers is not in the document]
Comment (chinokoronkwo) [reply package.json]: moved node-wifi to web node_modules

File: source/scenes/common/Login/Login.tsx [hunk 4]
```diff
@@ -27,6 +27,8 @@ import { useController } from 'hooks/index';
 
 // Imports
 import { schema } from './consts';
+import * as consts from './consts';
```
[the remark this answers is not in the document]
Comment (chinokoronkwo) [reply source/scenes/common/Login/Login.tsx]: remove const reference

File: source/scripts/Background/controllers/WalletController.ts [hunk 5]
```diff
@@ -65,10 +64,19 @@ export class WalletController implements IWalletController {
         catch (e) {
           return false;
         }
+        console.log('Login successful.');
```
[the remark this answers is not in the document]
Comment (chinokoronkwo) [reply source/scripts/Background/controllers/WalletController.ts]: removed

File: source/scripts/Background/controllers/WalletController.ts [hunk 6]
```diff
@@ -65,10 +64,19 @@ export class WalletController implements IWalletController {
...
-        await this.switchWallet(vault.activeWallet.id);
+        try {
+          await this.switchWallet(vault.activeWallet.id);
+        }
+        catch(e)
+        {
+          console.log('Error while switching wallet.');
+          console.log(e.toString());
+          return true;
+        }
```
[the remark this answers is not in the document]
Comment (chinokoronkwo) [reply source/scripts/Background/controllers/WalletController.ts]: moved opening inline, removed e.toString, returned false

File: source/scenes/common/Login/Login.tsx (hunk 2 above)
Comment (chinokoronkwo) [reply source/scenes/common/Login/Login.tsx]: changed all ids created to follow the following pattern: "<componentName>-<functionName>"

File: source/scenes/common/Login/Login.tsx (hunk 1 above)
Comment (chinokoronkwo) [reply source/scenes/common/Login/Login.tsx]: removed the state "Login successful." statement

**Pull request merged**